packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -263,6 +268,21 @@ const NavBar = (props: TNavbarProps) => {
     [props.project]
   );
 
+  const applicationLocale = props.applicationLocale;
+
+  if (props.isLoading || typeof applicationLocale === 'undefined') {
+    // Render the loading navbar as long as all the data
+    // hasn't been loaded, or if the project does not exist.
+    if (!props.isNewNavigationEnabledEvaluationReady) {
+      return null;
+    }
+    return props.isNewNavigationEnabled ? (
+      <NavBarSkeleton isExpanded={isMenuOpen} />
+    ) : (
+      <LoadingNavBar />
+    );
+  }
```

Comment:
```suggestion
  // Render the loading navbar as long as all the data
  // hasn't been loaded, or if the project does not exist.
  if (props.isLoading || typeof applicationLocale === 'undefined') {
    // Do not render anything until we have the feature flag remote value fetched so 
    // we avoid a flick effect in case it is 'true' (different value from the default one)
    if (!props.isNewNavigationEnabledEvaluationReady) {
      return null;
    }
    return props.isNewNavigationEnabled ? (
      <NavBarSkeleton isExpanded={isMenuOpen} />
    ) : (
      <LoadingNavBar />
    );
  }
```

Reply:
Thanks @CarlosCortizasCT! The change is here - [4f94eaa](https://github.com/commercetools/merchant-center-application-kit/pull/3165/commits/4f94eaac64d6f5fc4374bbc675e4d60e02316cc6).